glfw/x11_window.c

```diff
@@ -185,6 +185,7 @@ static int translateState(int state)
 {
     int mods = 0;
 
+    /* Need some way to expose hyper and meta without xkbcommon-x11 */
```

Comment:
Might be better to just take the current modifier state from xkb. However X.h has Mod5Mask as well, might be worth looking into.

Reply:
Mod5Mask is generally mapped to a level shift of some sort, in my understanding, which you rarely want to touch. Users of non-ASCII-like encodings may know better than I do. It's possible to check whether any keys are mapped to modifier bits, by going over all keys in the core X11 keyboard, and then using XLookupString to identify candidates for each of the modifiers. However, I'd be surprised if there exist non-XKB users of X11 who actually care about these modifiers. In other words, I'd wait for a bug report from at least one such user before implementing this.